### PR TITLE
cpu: aarch64: fix scratchpad memory initialization

### PR DESCRIPTION
**Pull request by aditew01**
This  fixes Conv dw (BF16) with Bias-dt=BF16. Bug introduced as part of this PR: https://github.com/uxlfoundation/oneDNN/pull/3441

I've added a fix and added unit testcases.

The failure was reported by unit test fails in this PR: https://github.com/pytorch/pytorch/pull/157994 

Example test : 

```
./benchdnn --conv --dt=bf16 --dir=FWD_I --bia-dt=bf16  g4mb2_ic4oc4_ih6oh4kh3sh1dh0ph0_iw6ow4kw3sw1dw0pw0
```

### CODE REVIEW

**Comment by vpirogov** [comment]
@aditew01, as this issue affects upstream PyTorch please open a backport to `rls-v3.9` branch.

**Comment by aditew01** [comment]
@vpirogov thanks for the prompt suggestion! raised a backport PR: https://github.com/uxlfoundation/oneDNN/pull/3766

**Review by Sqvid — APPROVED** [review]
Thank you for the fix!

**Comment by vpirogov** [comment]
make test

File: tests/benchdnn/inputs/conv/test_conv_depthwise [hunk 1]
```diff
@@ -16,3 +16,7 @@
 --batch=set_conv_dw
 --batch=shapes_dw_minibatch_2d-spatial
 --batch=shapes_dw_minibatch_channel_2d-spatial
+
+--reset --dt=bf16 --bia-dt=bf16
+--mb=2
+--dir=FWD_D,FWD_I --batch=shapes_mobilenet_dw
```
Comment (vpirogov) [inline tests/benchdnn/inputs/conv/test_conv_depthwise]: I'm a bit on the fence about these new testcases, as these will be executed on all supported platforms in perpetuity.

Comment (Sqvid) [reply tests/benchdnn/inputs/conv/test_conv_depthwise]: We could run just `--dir=FWD_D` to help a bit. We can also just run it against a single shape from `shapes_mobilenet_dw` instead of the batch. Would that be preferable?

Comment (aditew01) [reply tests/benchdnn/inputs/conv/test_conv_depthwise]: @vpirogov I see, I can add this reproducer which atleast guards the corner case not covered earlier. 
`./benchdnn --conv --dt=bf16 --dir=FWD_I --bia-dt=bf16  g4mb2_ic4oc4_ih6oh4kh3sh1dh0ph0_iw6ow4kw3sw1dw0pw0`

Comment (vpirogov) [reply tests/benchdnn/inputs/conv/test_conv_depthwise]: I think we have couple options:
* Stick specific case in question into [harness_conv_regression_general](https://github.com/uxlfoundation/oneDNN/blob/main/tests/benchdnn/inputs/conv/harness_conv_regression_general) set.
* Get input on general coverage extension from @dzarukin once he's back Monday.

File: src/cpu/aarch64/jit_uni_dw_conv_kernel_utils.hpp [hunk 2]
```diff
@@ -281,8 +281,11 @@ template <cpu_isa_t isa, data_type_t kernel_dt>
 void jit_uni_dw_conv_fwd_kernel_t<isa, kernel_dt>::init_scratchpad(
         memory_tracking::registrar_t &scratchpad, const jit_conv_conf_t &jcp) {
     using namespace dnnl::impl::memory_tracking::names;
-    if (jcp.with_bias && jcp.oc_without_padding != jcp.oc)
+    if (jcp.with_bias && jcp.bia_dt == data_type::bf16) {
+        scratchpad.book<float>(key_conv_bias_bf16_convert_wsp, jcp.oc);
```
Comment (dzarukin) [inline src/cpu/aarch64/jit_uni_dw_conv_kernel_utils.hpp:285]: What this scratchpad is needed for? Reading in the kernel with the appropriate data type should be available without extra memory...

File: tests/benchdnn/inputs/conv/test_conv_depthwise [hunk 3]
```diff
@@ -16,3 +16,5 @@
 --batch=set_conv_dw
 --batch=shapes_dw_minibatch_2d-spatial
 --batch=shapes_dw_minibatch_channel_2d-spatial
+
+--reset --dt=bf16 --bia-dt=bf16 --dir=FWD_I g4mb2_ic4oc4_ih6oh4kh3sh1dh0ph0_iw6ow4kw3sw1dw0pw0
```
Comment (dzarukin) [inline tests/benchdnn/inputs/conv/test_conv_depthwise]: I'd move it to a regression list with extra `_n"aarch64_jit_dw_bia_dt_bf16"` added to the descriptor.

Since `--bia-dt` is a relatively new flag for convolution, there's no coverage for it as it wasn't worked on.

File: src/cpu/aarch64/jit_uni_dw_conv_kernel_utils.hpp (hunk 2 above)
Comment (aditew01) [reply src/cpu/aarch64/jit_uni_dw_conv_kernel_utils.hpp:285]: Very valid. I think the current implementation relies on bias being fp32 and the scratch memory is initialized here: https://github.com/uxlfoundation/oneDNN/blob/7c4fe8411c9e620a844bf9482482717793d8c944/src/cpu/aarch64/jit_uni_dw_convolution.cpp#L55

**Comment by yanbing-j** [comment]
Hi @vpirogov @dzarukin, any other comments on this PR? Can this catch v3.9 patch release?

File: src/cpu/aarch64/jit_uni_dw_conv_kernel_utils.hpp (hunk 2 above)
Comment (dzarukin) [reply src/cpu/aarch64/jit_uni_dw_conv_kernel_utils.hpp:285]: Thanks for explaining. Resolving.

File: tests/benchdnn/inputs/conv/test_conv_regression [hunk 4]
```diff
@@ -15,3 +15,5 @@
 
 #
 --batch=harness_conv_regression_general
+
+--reset --dt=bf16 --bia-dt=bf16 --dir=FWD_I g4mb2_ic4oc4_ih6oh4kh3sh1dh0ph0_iw6ow4kw3sw1dw0pw0_n"aarch64_jit_dw_bia_dt_bf16"
```
Comment (dzarukin) [inline tests/benchdnn/inputs/conv/test_conv_regression]: I'm sorry I gave a wrong pointer in the beginning. Could you, please, put it under `harness_conv_regression_general`, it is the proper list to collect those single line cases for regressions.

**Review by dzarukin — APPROVED** [review]
